core/templates/domain/topic_viewer/read-only-topic-object.factory.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
```

Comment:
Hi, @Vinit-Dantkale I think this file should follow this name: `ReadyOnlyTopicObjectFactory.ts` (the spec file as well). All the factories files follow the same name pattern, you can check some examples in the codebase.

Reply:
It was `ReadyOnlyTopicObjectFactory.ts` , later name was changed as per [review](https://github.com/oppia/oppia/pull/8476#discussion_r375455286).

Reply:
Thanks, @Vinit-Dantkale for clarifying. @DubeySandeep can you explain why the file name has changed?

Reply:
Because we are trying to have a consistent pattern for filenames in the repository ie. we follow `<filename>.<file-type>.ts` for all the frontend files. (I had a discussion on this with bansalnatish on same and we had finalized this as the patter for the factory services file as well.)

Reply:
Ok, I got it. Do you intent to rename all the `*Factory.ts files` then? I'm just a little worried because I'm trying to find a pattern for frontend services in order to have a plan to achieve 100% coverage of it, and now I need to include this new pattern as well. I just want to make sure if this new pattern should be considered.

Reply:
@marianazangrossi, please do tell me as well

Reply:
Sure, thanks for clarifying @DubeySandeep